waku/factory/external_config.nim

```diff
@@ -749,10 +749,12 @@ proc parseCmdArg*(T: type EthRpcUrl, s: string): T =
   ## http://url/with/path
   ## http://url:port/path?query
   ## https://url:port/path?query
+  ## https://user:pass@url:port/path
+  ## http://username:password@url:port/path
   ## disallowed patterns:
   ## any valid/invalid ws or wss url
   var httpPattern =
-    re2"^(https?):\/\/([\w-]+(\.[\w-]+)*)(:[0-9]{1,5})?(\/[\w.,@?^=%&:\/~+#-]*)?$"
```

Comment:
I would recommended extracting that and adding some unit tests to get high confidence on correctness.